tensorflow/python/keras/layers/normalization.py

```diff
@@ -981,6 +982,31 @@ def __init__(self,
 
     self.supports_masking = True
 
+    # Indicates whether a faster fused implementation can be used. This will be
+    # set to True or False in build()"
+    self._fused = None
+
+  def _fused_can_be_used(self, ndims):
+    """Return false if fused implementation cannot be used.
+
+    Check if the axis is contiguous and can be collapsed into the last axis.
+    The self.axis is assumed to have no duplicates.
+    """
+    axis = sorted(self.axis)
+    can_use_fused = False
+
+    if axis[-1] == ndims-1 and axis[-1] - axis[0] == len(axis) - 1:
+      can_use_fused = True
+
+    # Fused layer norm may have numeric issues when epsilon < 1.001e-5 (see
+    # cudnn.h). Also, the gamma dtype needs to be float32 when using fused
```

Comment:
cudnn.h no longer has a requirement on the value of epsilon. Instead, say "fused_batch_norm will silently raise epsilon to be at least 1.001e-5, so we cannot used the fused version if epsilon is below that value."

Also, instead of checking gamma.dtype, just check if self.dtype != 'float32'. Note self.dtype is the variable dtype. Then you can also remove the float64 check in call(). You can update the comment, saying "Also, the variable dtype must be float32, as fused_batch_norm only supports float32 variables.